UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.Common/Helpers/AzureEventGridDomainService.cs

```diff
@@ -18,25 +18,59 @@ namespace UKHO.ExternalNotificationService.Common.Helpers
         {
             _logger.LogInformation(EventIds.CreateOrUpdateAzureEventDomainTopicStart.ToEventId(),
                     "Create or update azure event domain topic started for SubscriptionId:{SubscriptionId} with _D365-Correlation-ID:{correlationId} and _X-Correlation-ID:{CorrelationId} with Event domain topic {NotificationTypeTopicName}", subscriptionRequestMessage.SubscriptionId, subscriptionRequestMessage.D365CorrelationId, subscriptionRequestMessage.CorrelationId, subscriptionRequestMessage.NotificationTypeTopicName);
 
             EventGridManagementClient eventGridMgmtClient = await GetEventGridClient(_eventGridDomainConfig.SubscriptionId, cancellationToken);
             DomainTopic topic = await GetDomainTopic(eventGridMgmtClient, subscriptionRequestMessage.NotificationTypeTopicName, cancellationToken);
+            string eventSubscriptionScope = topic.Id;
+
+            string deadLetterDestinationResourceId = $"/subscriptions/{_eventGridDomainConfig.SubscriptionId}/resourceGroups/{_eventGridDomainConfig.ResourceGroup}/providers/Microsoft.Storage/storageAccounts/{_subscriptionStorageConfiguration.StorageAccountName}";
+
+            EventSubscription eventSubscription = new() {
+                Destination = new WebHookEventSubscriptionDestination()
+                {
+                    EndpointUrl = subscriptionRequestMessage.WebhookUrl
+                },
+                // The below are all optional settings
+                EventDeliverySchema = EventDeliverySchema.CloudEventSchemaV10,
+                /* Retry policy decides when an event can be marked as expired. 
+                   The default retry policy keeps the event alive for 24 hrs (=1440 mins or 30 retries with exponential backoffs)
+                   An event is marked as expired once any of the retry policy limits are exceeded. 
+                   Note: The below configuration for the retry policy will cause events to expire after one delivery attempt. 
+                   This is only to make it easier to help test/verify dead letter destinations quickly.
+                */
+                RetryPolicy = new RetryPolicy()
+                {
+                    MaxDeliveryAttempts = _eventGridDomainConfig.MaxDeliveryAttempts,
+                    EventTimeToLiveInMinutes = _eventGridDomainConfig.EventTimeToLiveInMinutes,
+                },
+                // With dead-letter destination configured, all expired events will be delivered to this destination.
+                // Note: only Storage Blobs are supported as dead letter destinations as of now.
+                DeadLetterDestination = new StorageBlobDeadLetterDestination()
+                {
+                    ResourceId = deadLetterDestinationResourceId,
+                    BlobContainerName = _subscriptionStorageConfiguration.StorageContainerName,
+                }
+            };
```

Comment:
This is importatnt business logic and really should be covered by unit tests but it's in a class that is marked as `ExcludeFromCodeCoverage`.

Could it be be refactored into a helper class that does have unit tests please?

Reply:
Sorry @Siddartha-tech, the business logic here is not just the getting of the subscription values from config.

What needs testing is that we construct an `EventSubscription` object which has the correct values for:
- `Destination`
- `EventDeliverySchema`
- `RetryPolicy`
- `DeadLetterDestination`

Happy to connect if you're not sure what I mean.

Reply:
Hi, as discussed try to compare fake object for "Destination" but giving exception as follow
Expected: <Microsoft.Azure.Management.EventGrid.Models.WebHookEventSubscriptionDestination>
But was: <Microsoft.Azure.Management.EventGrid.Models.WebHookEventSubscriptionDestination>
So, used only 'instance of'